RATIONALE.md

```diff
@@ -25,6 +25,21 @@ At some point, we may allow extensions to supply their own platform-specific
 hooks. Until then, one end user impact/tradeoff is some glitches trying
 untested platforms (with the Compiler runtime).
 
+### Why do we use CGO to implement system calls on darwin?
+
+wazero is dependency and CGO free by design. In some cases, we have code that
+can optionally use CGO, but retain a fallback for when that's disabled. The only
+operating system (`GOOS`) we use CGO by default in is `darwin`.
+
+Unlike other operating systems, regardless of `CGO_ENABLED`, Go always uses
+"CGO" mechanisms in the runtime layer of `darwin`, as that's where Apple
```

Comment:
found the reference to this https://developer.apple.com/library/archive/qa/qa1118/_index.html#//apple_ref/doc/uid/DTS10001666

thanks to the perplexity.ai :D https://www.perplexity.ai/?s=u&uuid=d06fc67f-c472-4540-b572-8ee9bad40acf

Reply:
quote: 

> Apple does not support statically linked binaries on Mac OS X. A statically linked binary assumes binary compatibility at the kernel system call interface, and we do not make any guarantees on that front. Rather, we strive to ensure binary compatibility in each dynamically linked system library and framework.

Reply:
thanks, added the quote!